src/main/scala/dpla/ingestion3/harvesters/api/ApiHarvester.scala

```diff
@@ -36,8 +38,13 @@ abstract class ApiHarvester(shortName: String,
     * This is lazy b/c queryParams should be printed before avroWriter is set.
     * @see doHarvest
     */
-  protected lazy val avroWriter: DataFileWriter[GenericRecord] =
-  getAvroWriter(outputFile, schema)
+  protected lazy val avroWriter: DataFileWriter[GenericRecord] = {
+    val dirName = if (outputDir.endsWith("/")) outputDir.dropRight(1) else outputDir
+    val fileName = dirName + "/api_harvest.avro"
+    val dir = new File(dirName).mkdir
```

Comment:
What about using `mkdirs` instead of `mkdir` to handle the case where multiple parent directories do not already exist?

Reply:
Excellent point.